CarRentals/CarRentalsWebAPI/Controllers/CarsController.cs

```diff
@@ -37,19 +38,32 @@ public async Task<ActionResult<Car>> GetCar(int id)
                 return NotFound();
             }
 
-            return car;
+            return new CarDto(car);
         }
 
         // PUT: api/Cars/5
         [HttpPut("{id}")]
-        public async Task<IActionResult> PutCar(int id, Car car)
+        public async Task<IActionResult> PutCar(int id, CarDto carDto)
         {
-            if (id != car.Id)
+            if (id != carDto.Id)
             {
                 return BadRequest();
             }
 
-            _context.Entry(car).State = EntityState.Modified;
+            var carModel = await _context.Cars.FindAsync(id);
+
+            if (carModel == null)
+            {
+                return NotFound();
+            }
+
+            carModel.Id = carDto.Id;
+            carModel.Doors = carDto.Doors;
+            carModel.Brand = BrandDto.DtoToEntity(carDto.Brand);
+            carModel.Color = carDto.Color;
+            carModel.IsRented = carDto.IsRented;
+            carModel.Transmition = carDto.Transmition;
+            carModel.Model = carDto.Model;
```

Comment:
@jrocaspana I have modified many things according to your comments. 
So, I did this method in relation to "the bible" Microsoft Documentation. 
According with your comment I should equalize ids instead of return BadRequest(). But I don't understand why I would have to do that because it is an id validation.

I belive that is correct the rest of the code, but please you let me know.

Reply:
your path `[HttpPut("{id}")]` is stating you will update the object with `{id}` in the database with whatever thing you are sending in the body (a CardDto in this case). It's a common thing to have "update DTOs" that are your current class WITHOUT the `Id` and/or any other property that it's readonly.
making both IDs equal is not strictly necessary, it depends of the underlying implementation (whether your services/repositories also expect the `id, object` pair or only the `object` and they obtain the `id` from it)

as a side note:  `carModel.Id = carDto.Id;` this is never done, we don't mess with DB primary keys in code 😄 

let me know and we can arrange a quick call to talk about this!

Reply:
also, before working on the service layers, let's merge this PR so we keep changes under control! good work!!

Reply:
Perfect Jose! Thank you very much for your feedback. I would like to talk with you a about some cuestions of the api. I want to know if I understood well to continue with the next step. Let me know if you can after 4pm.

Reply:
today at 4pm works for you? I have a 30 minutes time window

Reply:
Yes it’s perfect!